### DISCUSSION TITLE
Use watch systemd service with existing docker container?

### DISCUSSION THREAD

**Discussion by SawkeeReemo** [Taxel/PlexTraktSync#1961] [Q&A]
I'm a little new to creating systemd services on Linux (Ubuntu LTS 22.04, specifically). 
I currently have two docker containers built labeled:
`plextraktsync_sync`
`plextraktsync_watch`

As should be obvious, the first one does a full sync, and the second one runs a watch scrobble. I would like to switch over to using the systemd service to make scrobbling more immediate, but I'm unsure of how to execute the command in the service file. [Looking at the service file from the main page](https://github.com/Taxel/PlexTraktSync?tab=readme-ov-file#systemd-setup), my assumption is that I need to change the:

```
[Service]
ExecStart=plextraktsync watch
```
to:
```
[Service]
ExecStart=/usr/bin/docker compose run --rm plextraktsync watch
```

Is that correct?

**Comment by glensc** [comment]
what do you mean labeled? describe everything you're doing, leaving no room to guess. why just do it, why asking? you should share error you're getting.

also unclear what you mean by making immediate, what's problem with other setup? what is the other setup?

**Reply by SawkeeReemo** [reply]
Sorry, “labeled” was not the right word. I have two docker containers for PlexTraktSync. One is named `plextraktsync_sync` and the other is called `plextraktsync_watch`. 

_sync runs a sync command
_watch runs a watch command

I was just asking how to set up the systemd service since I use these in docker containers and not installed directly on the host. 

Is what I wrote above for the service correct? I’m new to writing a system service, so I don’t really know what I’m doing there. 

There are no errors or anything, I’m just trying to follow the directions on the main page to set up PlexTraktSync to scrobble immediately instead of waiting for my scheduled run of the docker container. But the README doesn’t really say how to run this as a system service when you’re using a docker container. Just looking for clarification.

**Reply by glensc** [reply]
if there are no errors, what are you even asking?

the one certain problem I see in your config is that you need to change working directory for docker-compose.yml to be located.

```
[Service]
WorkingDirectory=/usr/local
```

or if you already have containers running, use `docker exec <containername> plextraktsync <thing>`, this one doesn't need to change working directory.
